platform/vs/tests/bgp/test_default_route.py

```diff
@@ -0,0 +1,48 @@
+from swsscommon import swsscommon
+import os
+import re
+import time
+import json
+import pytest
+
+def test_DefaultRoute(dvs, testlog):
+
+    dvs.copy_file("/etc/frr/", "bgp/files/default_route/bgpd.conf")
+    dvs.runcmd("supervisorctl start bgpd")
+    dvs.runcmd("ip addr add 10.10.10.0/31 dev Ethernet0")
+    dvs.runcmd("config interface startup Ethernet0")
+    dvs.runcmd("ip route del 0.0.0.0/0")
+    dvs.runcmd("vtysh -c \"confgure terminal\" -c \"ip route 0.0.0.0/0 via 172.17.0.1 200\"")
+
+    dvs.servers[0].runcmd("ip addr add 10.10.10.1/31 dev eth0")
+    dvs.servers[0].runcmd("ifconfig eth0 up")
+
+    time.sleep(5)
+
+    print(dvs.runcmd("supervisorctl status"))
+
+    p = dvs.servers[0].runcmd_async("exabgp -d bgp/files/default_route/default_route.conf")
+
+    time.sleep(10)
+
+    (exit_code, output) = dvs.runcmd(["redis-cli", "hgetall", "ROUTE_TABLE:0.0.0.0/0"])
+    print(exit_code, output)
+
+    # make sure 10.10.10.1 is the correct next hop for default route
+    assert "10.10.10.1" in output
+
+    # insert default route for table default
+    dvs.runcmd("ip route add default via 172.17.0.1 table default")
```

Comment:
`ip route add` installs a kernel route to zebra. The administrative distance of the kernel route is 0 since the metric value is 0 and zebra uses the upper byte of kernel route metric as the distance. So the kernel route wins the BGP route and the only nexthop will stay eth0 until the kernel route is deleted. It's an expected behaviour and will prevent all subsequent BGP updates from taking effect in swss, and the ASIC. Filtering routes with eth0 nexthop just hides the problem and won't fix it.

Reply:
Just to clarify, these are two different tables - `default `and `main`. In Sonic, eth0 routes are part of `default route table` and we would want to give preference to BGP routes in swss and ASIC (which is part of main table) and not the eth0 routes (kernel). I think your point is valid if we have all routes in a single routing table. Did we miss anything?

Reply:
Thanks. The two difference tables you pointed out are what I've missed. And I've successfully reproduced it in our environment.